client_code/pedantic.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 anvilistas
+
+__version__ = "0.0.1"
+
+
+class BaseValidator:
+    value = None
+
+    def is_valid(self):
+        raise NotImplementedError
+
+    def __get__(self, obj):
+        return self.value
+
+    def __set__(self, obj, value):
+        self.value = value
+        if not self.is_valid():
+            raise ValueError(f"{value} is not a valid value")
+
+
+def add_validator(cls, attr, validator):
+    cls.validators.append(validator.is_valid)
+    setattr(cls, attr, validator)
```

Comment:
We probably want to know if the cls already had this attribute as a property or some other getter and setter. Maybe passing oldget and oldset to the Validator which could default to getting and setting from `self.__dict__`.

Reply:
I'd be minded to raise an error in that case.